test/api/ticket/api_test.go

```diff
@@ -0,0 +1,154 @@
+package ticket
+
+import (
+	"testing"
+
+	"github.com/konveyor/tackle2-hub/api"
+	"github.com/konveyor/tackle2-hub/test/assert"
+)
+
+func TestTicketCRUD(t *testing.T) {
+	for _, r := range Samples {
+		t.Run("Ticket "+r.Kind+" CRUD", func(t *testing.T) {
+
+			// Create a sample Application for the ticket.
+			app := api.Application{
+				Name: r.Application.Name,
+			}
+			assert.Must(t, Application.Create(&app))
+
+			// Create a sample identity for the tracker
+			identity := api.Identity{
+				Name: "Sample Identity",
+				Kind: "Sample Kind",
+			}
+			assert.Must(t, Identity.Create(&identity))
+
+			// create a sample tracker for the ticket.
+			tracker := api.Tracker{
+				Name: r.Tracker.Name,
+				URL:  "www.sample.com",
+				Kind: "jira-onprem",
+				Identity: api.Ref{
+					ID:   1,
+					Name: "Sample Identity",
+				},
+			}
+			assert.Must(t, Tracker.Create(&tracker))
+
+			// Create a sample ticket
+			assert.Must(t, Ticket.Create(&r))
+
+			// Get.
+			got, err := Ticket.Get(r.ID)
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Compare got values with expected values.
+			AssertEqualTickets(t, got, r)
+
+			// Delete ticket and its related resources.
+			assert.Must(t, Ticket.Delete(r.ID))
+			assert.Must(t, Tracker.Delete(tracker.ID))
+			assert.Must(t, Identity.Delete(identity.ID))
+			assert.Must(t, Application.Delete(app.ID))
+
+			// Check if the Ticket is present even after deletion or not.
+			_, err = Ticket.Get(r.ID)
+			if err == nil {
+				t.Errorf("Resource exits, but should be deleted: %v", r)
+			}
+		})
+	}
+}
+
+func TestTicketList(t *testing.T) {
+	createdTickets := []api.Ticket{}
+
+	for _, r := range Samples {
+		// Create a sample Application for the ticket.
+		app := api.Application{
+			Name: r.Application.Name,
+		}
+		assert.Must(t, Application.Create(&app))
+
+		// Create a sample identity for the tracker
+		identity := api.Identity{
+			Name: "Sample Identity",
+			Kind: "Sample Kind",
+		}
+		assert.Must(t, Identity.Create(&identity))
+
+		// create a sample tracker for the ticket.
+		tracker := api.Tracker{
+			Name: r.Tracker.Name,
+			URL:  "www.sample.com",
+			Kind: "jira-onprem",
+			Identity: api.Ref{
+				ID:   1,
+				Name: "Sample Identity",
+			},
+		}
+		assert.Must(t, Tracker.Create(&tracker))
+
+		// Create a sample ticket
+		assert.Must(t, Ticket.Create(&r))
+		createdTickets = append(createdTickets, r)
+	}
+
+	// List Tickets.
+	got, err := Ticket.List()
```

Comment:
There two additional filters in List actions - by application and tracker - https://github.com/konveyor/tackle2-hub/blob/main/api/ticket.go#L74-L79. Maybe it is not required for simple test, but might be worth for adding those.